### PR TITLE
docs: spec cleanups and clarifications

### PR DESCRIPTION
**Pull request by ss2165**

### CODE REVIEW

File: specification/hugr.md [hunk 1]
```diff
@@ -533,50 +524,28 @@ may be a `FuncDefn`, `TailLoop`, `DFG`, `Case` or `DFB` node.
...
-    extension, taking a graph argument; and `run_circuit` will return the
-    same way.
+  - `Local`: Source and target nodes must have the same parent.
+  - `Ext`: Edges "in" from a dataflow ancestor.
+  - `Dom`: Edges from a dominating basic block in a control-flow graph.
 
-#### **Non-local Edges**
 
-**For ``CopyableType`` values only** we allow dataflow edges (i.e. both Value and Static)
+We allow non-local dataflow edged
```
Comment (cqc-alec) [inline specification/hugr.md:539]:
```suggestion
We allow non-local dataflow edges
```

File: specification/hugr.md [hunk 2]
```diff
@@ -1004,23 +1004,27 @@ There are three classes of type: ``AnyType`` $\supset$ ``CopyableType`` $\supset
 
     In fully qubit-counted contexts programs take in a number of qubits as input and return the same number, with no discarding. See [quantum extension](#quantum-extension) for more.
 
-  - The next class is ``CopyableType``, i.e. types holding ordinary classical data, where values can be copied (and discarded, the 0-ary copy). This allows multiple (or 0) outgoing edges from an outport; also these types can be sent down Static edges.
+  - The next class is ``CopyableType``, i.e. types holding ordinary classical
+    data, where values can be copied (and discarded, the 0-ary copy). This
+    allows multiple (or 0) outgoing edges from an outport; also these types can
+    be sent down Static edges. Note dataflow inputs (Value and Static) always
+    require a single a connection.
```
Comment (cqc-alec) [inline specification/hugr.md:1011]:
```suggestion
    require a single connection.
```

File: specification/hugr.md [hunk 3]
```diff
@@ -1004,23 +1004,27 @@ There are three classes of type: ``AnyType`` $\supset$ ``CopyableType`` $\supset
 
     In fully qubit-counted contexts programs take in a number of qubits as input and return the same number, with no discarding. See [quantum extension](#quantum-extension) for more.
 
-  - The next class is ``CopyableType``, i.e. types holding ordinary classical data, where values can be copied (and discarded, the 0-ary copy). This allows multiple (or 0) outgoing edges from an outport; also these types can be sent down Static edges.
+  - The next class is ``CopyableType``, i.e. types holding ordinary classical
+    data, where values can be copied (and discarded, the 0-ary copy). This
+    allows multiple (or 0) outgoing edges from an outport; also these types can
+    be sent down Static edges. Note dataflow inputs (Value and Static) always
```
Comment (cqc-alec) [inline specification/hugr.md:1010]:
```suggestion
    be sent down Static edges. Note: dataflow inputs (Value and Static) always
```

File: specification/hugr.md [hunk 4]
```diff
@@ -1084,50 +1088,35 @@ which stands in for a row. Hence, when checking the inputs and outputs
...
-### Types of built-ins
-
-We will provide some built in modules to provide basic functionality.
-We will define them in terms of extensions. We have the “builtin”
-extension which should always be available when writing hugr plugins.
-This includes Conditional and TailLoop nodes, and nodes like `Call`:
+### Rewriting Extension Requirements
 
-$\displaystyle{\frac{\mathrm{args} : [R] \vec{I}}{\textbf{call} \langle \textbf{Function}[R](\vec{I}, \vec{O}) \rangle (\mathrm{args}) : [R] \vec{O}}}$
+Extensions requirements help denote different runtime capabilities.
```
Comment (cqc-alec) [inline specification/hugr.md:1093]:
```suggestion
Extension requirements help denote different runtime capabilities.
```

File: specification/hugr.md [hunk 5]
```diff
@@ -1746,14 +1734,14 @@ that *v* is lifted to have extension requirement
 R so that it matches the type
 of input to the next iterations of the loop.
 
-$\displaystyle{\frac{\Theta : [R] \textbf{Function}[R](\vec{X}, \vec{Y}) \quad \vec{x} : [R] \vec{X}}{\textbf{call\\_indirect}(\Theta, \vec{x}) : [R] \vec{Y}}}$
+$\displaystyle{\frac{\Theta : [R] \textbf{Function}[R](\vec{X}, \vec{Y}) \quad \vec{x} : [R] \vec{X}}{\textbf{call\_indirect}(\Theta, \vec{x}) : [R] \vec{Y}}}$
 
 **CallIndirect** - This has the same feature as **loop**: running a
 graph requires it’s extensions.
```
Comment (cqc-alec) [inline specification/hugr.md:1740]:
```suggestion
graph requires its extensions.
```
(drive-by)

File: specification/hugr.md [hunk 6]
```diff
@@ -1746,14 +1734,14 @@ that *v* is lifted to have extension requirement
 R so that it matches the type
 of input to the next iterations of the loop.
 
-$\displaystyle{\frac{\Theta : [R] \textbf{Function}[R](\vec{X}, \vec{Y}) \quad \vec{x} : [R] \vec{X}}{\textbf{call\\_indirect}(\Theta, \vec{x}) : [R] \vec{Y}}}$
+$\displaystyle{\frac{\Theta : [R] \textbf{Function}[R](\vec{X}, \vec{Y}) \quad \vec{x} : [R] \vec{X}}{\textbf{call\_indirect}(\Theta, \vec{x}) : [R] \vec{Y}}}$
```
Comment (cqc-alec) [inline specification/hugr.md:1737]: The double backslash is necessary: see https://github.com/CQCL/hugr/blob/spec/cleanups/specification/hugr.md .

File: specification/hugr.md [hunk 7]
```diff
@@ -1746,14 +1734,14 @@ that *v* is lifted to have extension requirement
...
 
 **CallIndirect** - This has the same feature as **loop**: running a
 graph requires it’s extensions.
 
-$\displaystyle{\frac{}{\textbf{to\\_const} \langle \textbf{Function}[R](\vec{I}, \vec{O}) \rangle (\mathrm{name}) : [\emptyset] \textbf{Function}[R](\vec{I}, \vec{O})}}$
+$\displaystyle{\frac{}{\textbf{load\_const} \langle \textbf{Function}[R](\vec{I}, \vec{O}) \rangle (\mathrm{name}) : [\emptyset] \textbf{Function}[R](\vec{I}, \vec{O})}}$
```
Comment (cqc-alec) [inline specification/hugr.md:1742]: Keep double-backslash.

**Review by cqc-alec — CHANGES_REQUESTED** [review]

File: specification/hugr.md (hunk 6 above)
Comment (ss2165) [reply specification/hugr.md:1737]: ah thanks, looks like my local renderer and github disagree

File: specification/hugr.md (hunk 7 above)
Comment (ss2165) [inline specification/hugr.md:1742]:
```suggestion
$\displaystyle{\frac{}{\textbf{load\\_const} \langle \textbf{Function}[R](\vec{I}, \vec{O}) \rangle (\mathrm{name}) : [\emptyset] \textbf{Function}[R](\vec{I}, \vec{O})}}$
```

File: specification/hugr.md (hunk 6 above)
Comment (ss2165) [inline specification/hugr.md:1737]:
```suggestion
$\displaystyle{\frac{\Theta : [R] \textbf{Function}[R](\vec{X}, \vec{Y}) \quad \vec{x} : [R] \vec{X}}{\textbf{call\\_indirect}(\Theta, \vec{x}) : [R] \vec{Y}}}$
```

**Review by cqc-alec — APPROVED** [review]
